### PR TITLE
Hotfix/register animal

### PR DESCRIPTION
**Pull request by jhagberg**
Managers can now sell animals to GX1 and MX1. 

Searching for breedings does no longer render an error if breeding_date i None

### CODE REVIEW

File: app/utils/database.py [hunk 1]
```diff
@@ -970,7 +970,7 @@ def can_edit(self, identifier):
         `identifier` will be parsed as:
 
         `^([a-zA-Z][0-9]+-[0-9]+)$`: individual
-        `^([a-zA-Z][0-9]+)$`: herd
+        `^(([G|M]X1)|[a-zA-Z][0-9]+)$`: herd
```
Comment (andkaha) [inline app/utils/database.py]: The regular expression `[G|M]` matches one of the three characters `G`, `|`, or `M`.  You want to use `[GM]`, or possibly just `.` (dot) to match any single character before `X1`.

File: app/utils/database.py [hunk 2]
```diff
@@ -1005,7 +1005,7 @@ def can_edit(self, identifier):
             except DoesNotExist:
                 pass
 
-        elif re.match("^([a-zA-Z][0-9]+)$", identifier):
+        elif re.match("^(([G|M]X1)|[a-zA-Z][0-9]+)$", identifier):
```
Comment (andkaha) [inline app/utils/database.py]: Here too, the regular expression could use `.` in place of the faulty `[G|M]`.

**Review by andkaha — CHANGES_REQUESTED** [review]
Error in regular expressions.

File: app/utils/database.py (hunk 1 above)
Comment (jhagberg) [reply app/utils/database.py]: Aha I am mixing Javascript good catch, Yes should be GM instead

Comment (jhagberg) [inline app/utils/database.py]:
```suggestion
        `^(([GM]X1)|[a-zA-Z][0-9]+)$`: herd
```

File: app/utils/database.py (hunk 2 above)
Comment (jhagberg) [inline app/utils/database.py]:
```suggestion
        elif re.match("^(([GM]X1)|[a-zA-Z][0-9]+)$", identifier):
```

**Review by andkaha — APPROVED** [review]
Good now.

**Pull request merged**